code/modules/food/cooking_machines.dm

```diff
@@ -477,6 +477,88 @@ var/global/ingredientLimit = 10
 	takeIngredient(I, L, TRUE) //shove the item in, even if it can't be deepfried normally
 	empty_icon()
 
+
+// Sugarizer ///////////////////////////////////////
+// its like a deepfrier
+
+// but with sugar
+
+#define SUGARIZER_MINSUGAR 50
+
+/obj/machinery/cooking/deepfryer/sugarizer
+	name = "sugarizer"
+	desc = "Creates sugar copies of stuff."
+	icon_state = "sugarizer_off"
+	icon_state_on = "sugarizer_on"
+	foodChoices = null
+	cookTime = 100
+	recursive_ingredients = 1
+	cks_max_volume = 400
+	cooks_in_reagents = 1
+
+/obj/machinery/cooking/deepfryer/sugarizer/validateIngredient(var/obj/item/I, var/force_cook)
+	if(I.w_class < W_CLASS_LARGE)
+		. = "valid"
+
+	else
+		. = "The sugarizer will not be able to replicate that."
+	if((. == "valid") && (!foodNesting))
+		if(findtext(I.name,"sugar"))
```

Comment:
Can't you just check if it's a `obj/item/weapon/reagent_containers/food/snacks/deepfryholder`?  That'd forbid sugarizing deep-fried things and vice versa, but it seems like the simplest solution to prevent recursion.